token/js/src/extensions/transferHook/state.ts

```diff
@@ -49,3 +52,64 @@ export function getTransferHookAccount(account: Account): TransferHookAccount |
         return null;
     }
 }
+
+/** ExtraAccountsMeta as stored by the transfer hook program */
+export interface ExtraAccountsMeta {
+    discriminator: number;
+    addressConfig: Uint8Array;
+    isSigner: boolean;
+    isWritable: boolean;
+}
+
+/** Buffer layout for de/serializing an ExtraAccountsMeta */
+export const ExtraAccountsMetaLayout = struct<ExtraAccountsMeta>([
+    u8('discriminator'),
+    blob(32, 'addressConfig'),
+    bool('isSigner'),
+    bool('isWritable'),
+]);
+
+export interface ExtraAccountsMetaList {
+    count: number;
+    extraAccounts: ExtraAccountsMeta[];
+}
+
+/** Buffer layout for de/serializing a list of ExtraAccountsMeta prefixed by a u32 length */
+export const ExtraAccountsMetaListLayout = struct<ExtraAccountsMetaList>([
+    u32('count'),
+    seq<ExtraAccountsMeta>(ExtraAccountsMetaLayout, greedy(ExtraAccountsMetaLayout.span), 'extraAccounts'),
+]);
+
+/** Unpack an extra account metas account and parse the data into a list of ExtraAccountMetas */
+export function getExtraAccountMetas(account: AccountInfo<Buffer> | null): ExtraAccountsMeta[] | null {
+    if (account == null) {
+        throw null;
+    }
+    return ExtraAccountsMetaListLayout.decode(account.data).extraAccounts;
+}
```

Comment:
It doesn't look like these bad boys need to be exported. Can prob keep them private

Reply:
My thinking for these ones being exposed was that potentially someone (probably wallet software) would want to decode the ExtraAccountMeta account manually if they do not want to use `addExtraAccountsToInstruction` which is async.

One such case could be because when someone is fetching multiple accounts using `Connection.getMultipleAccountsInfo` instead of fetching each account sequentially.

Now that I think of it they could also use `getExtraAccountMetas` and `resolveExtraAccountMeta` directly which makes more sense.

Only thing you can do with `ExtraAccountsMetaListLayout` that you cannot `getExtraAccountMetas` is encode the data. This might be nice to have for e2e testing purposes. For example to set arbitrary accounts when using a tool like `solana-bankrun`.

Reply:
yeah I guess that's fair. After all you can just use the Rust types directly as well. Carry on mate!